### DISCUSSION TITLE
Error with my folder name "zlib"

### DISCUSSION THREAD

**Discussion by lucky33newman** [pytest-dev/pytest#10376] [Q&A] [answered]
Hi guys,

I am meeting a strange failure.
--------------------------------------------------------------------------------------------
ImportError while importing test module '/usr/AGL/agl-test/tests/zlib/test_zlib.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/usr/lib/python3.10/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
E   ModuleNotFoundError: No module named 'zlib.test_zlib'; 'zlib' is not a package
--------------------------------------------------------------------------------------------

It is quite common import error? I do not think so, because the folder ''/usr/AGL/agl-test/tests/zlib''
it can be any other name, just not 'zlib', for example, other test suite like "expat", "bzip2", their folder
was named like "/usr/AGL/agl-test/tests/expat" and "/usr/AGL/agl-test/tests/bzip2", everything works
fine, also if I change the folder name "/usr/AGL/agl-test/tests/zlib" to "/usr/AGL/agl-test/tests/zlib_test"
(just example, it does not have to be fixed like this), the Error Message disappears, why?

Help me and I will thank you :)

**Comment by RonnyPfannschmidt** [comment]
This is a unfortunate quirk of having to add seeming import roots to sys. Path, the zlib folder conflicts with the stdlib that way

**Comment by lucky33newman** [comment, marked as answer]
Thanks, I think It is the cause.

It can be solved, by changing the folder name.

But if I do not change the folder name. How may I avoid such conflict?

I know It is stupid to name the folder with a dangerous name 'zlib'. But my
boss decided it's gonna be just like this. :(

Any fancy solutions?

Ronny Pfannschmidt ***@***.***> 于 2022年10月13日周四 14:09写道：

**Reply by RonnyPfannschmidt** [reply]
Add a few `__init__.py` files to trick pytest into adding something else to sys. Path

**Comment by lucky33newman** [comment]
Oh yes. :)

 __init__.py helps.

Much thanks.


Ronny Pfannschmidt ***@***.***> 于 2022年10月14日周五 00:26写道：
